x-pack/plugin/index-lifecycle/src/main/java/org/elasticsearch/xpack/indexlifecycle/IndexLifecycle.java

```diff
@@ -129,18 +132,26 @@ protected Clock getClock() {
     public List<RestHandler> getRestHandlers(Settings settings, RestController restController, ClusterSettings clusterSettings,
             IndexScopedSettings indexScopedSettings, SettingsFilter settingsFilter, IndexNameExpressionResolver indexNameExpressionResolver,
             Supplier<DiscoveryNodes> nodesInCluster) {
+        if (enabled == false) {
+            return emptyList();
+        }
         return Arrays.asList(
                 new RestPutLifecycleAction(settings, restController),
                 new RestGetLifecycleAction(settings, restController),
-                new RestDeleteLifecycleAction(settings, restController));
+                new RestDeleteLifecycleAction(settings, restController),
+                new RestMoveToStepAction(settings, restController));
     }
 
     @Override
     public List<ActionHandler<? extends ActionRequest, ? extends ActionResponse>> getActions() {
+        if (enabled == false) {
```

Comment:
I've added this since `@Inject`ing the IndexLifecycleService when not enabled blows up Guice initialization